packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/PaymentMethods/Methods/index.tsx

```diff
@@ -0,0 +1,318 @@
+      case 'BANK_TRANSFER':
+      case 'LINK_BANK':
+        return <Image name='bank' height='48px' />
+      case 'BANK_ACCOUNT':
+        return (
+          <IconContainer>
+            <Icon size='18px' color='blue600' name='arrow-down' />
+          </IconContainer>
+        )
+      case 'PAYMENT_CARD':
+        return (
+          <IconContainer>
+            <Icon size='18px' color='blue600' name='credit-card-sb' />
+          </IconContainer>
+        )
+      case 'USER_CARD':
+        const { card } = value
+        if (!card) {
+          return <></>
+        }
+        const cardType = CARD_TYPES.find(cc => cc.type === card.type)
+        return (
+          <img
+            height='18px'
+            width='auto'
+            src={cardType ? cardType.logo : DEFAULT_CARD_SVG_LOGO}
+          />
+        )
+      case 'FUNDS':
+        return (
+          <Icon
+            size='32px'
+            color='fiat'
+            name={value.currency.toLowerCase() as 'eur' | 'gbp'}
+          />
+        )
+      default:
+        return <Image name='blank-card' />
+    }
+  }
+
+  renderBankText = (value: SBPaymentMethodType): string => {
+    return value.details
+      ? value.details.accountName
+        ? value.details.accountName
+        : value.details.accountNumber
+      : 'Bank Account'
+  }
+
+  renderCardText = (value: SBPaymentMethodType): string => {
+    return value.card
+      ? value.card.label
+        ? value.card.label
+        : value.card.type
+      : 'Credit or Debit Card'
+  }
+
+  render () {
+    const { fiatCurrency, orderType } = this.props
+    const availableCards = this.props.cards.filter(
+      card => card.state === 'ACTIVE' && orderType === 'BUY'
+    )
+
+    const defaultMethods = this.props.paymentMethods.methods.map(value => ({
+      text: this.getType(value),
+      value
+    }))
+
+    const defaultCardMethod = this.props.paymentMethods.methods.find(
+      m => m.type === 'PAYMENT_CARD' && orderType === 'BUY'
+    )
+
+    const funds = defaultMethods.filter(
+      method =>
+        method.value.type === 'FUNDS' &&
+        method.value.currency in WalletFiatEnum &&
+        (orderType === 'SELL' ||
+          Number(
+            this.props.balances[method.value.currency as WalletCurrencyType]
+              ?.available
+          ) > 0)
+    )
+
+    const paymentCard = defaultMethods.find(
+      method => method.value.type === 'PAYMENT_CARD' && orderType === 'BUY'
+    )
+    const bankAccount = defaultMethods.find(
+      method => method.value.type === 'BANK_ACCOUNT' && orderType === 'BUY'
+    )
+    const bankTransfer = defaultMethods.find(
+      method => method.value.type === 'BANK_TRANSFER' && orderType === 'BUY'
+    )
+
+    const cardMethods = availableCards.map(card => ({
+      text: card.card
+        ? card.card.label
+          ? card.card.label
+          : card.card.type
+        : 'Credit or Debit Card',
+      value: {
+        ...card,
+        card: card.card,
```

Comment:
I think `card: card.card` is included in `...card`

Reply:
Funny but not aways :( this is a definitely thing to be improved.